### ISSUE TITLE
Preview Not Working for Next.js Project Using SliceZone

### ISSUE THREAD

**Issue by peroh** [prismicio/slice-machine#68]
### Version

- Prismic CLI version (know it by running `prismic --version`):  3.8.3-beta.0

### Reproduction

- Create a next.js project configured with slice machine
- Use `SliceZone` component from `next-slicezone` in a dynamically generated page route
- Set up prismic previews
- Deploy to Vercel
- Attempt to preview on deployed site

e.g. `pages/[uid].js`:

```js
import SliceZone from "next-slicezone";
import { useGetStaticPaths } from "next-slicezone/hooks";
import Head from "next/head";
import Layout from "../../components/layout";
import { useGetStaticProps } from "../../next-slicezone/hooks";
import { Client } from "../../prismic";
import resolver from "../../sm-resolver.js";

export default function Page(props) {
  return (
    <Layout>
        <Head>
          <title>Test Page</title>
        </Head>
        <SliceZone {...props} resolver={resolver} />
    </Layout>
  );
}

export const getStaticPaths = useGetStaticPaths({
  client: Client(),
  formatPath: ({ uid }) => ({ params: { uid } }),
});

export const getStaticProps = useGetStaticProps({
  client: Client(),
  uid: ({ params }) => params.uid,
});
```

### What is expected?

I should be able to preview a prismic page and see the updated `SliceZone` components when deployed to Vercel.

### What is actually happening?

The preview works locally, but when deployed the `SliceZone` component does not render anything.

### Any additional information

The problem seems to lie in `useGetStaticProps` from `next-slicezone/hooks`. When deployed to Vercel, `useGetStaticProps` makes a call to `next-slicezone/features/registry.js`, and Vercel cannot find the config file [here](https://github.com/prismicio/slice-machine/blob/master/packages/next-slicezone/features/registry.js#L105).

When deployed, `pathToSmFile` is `/var/task/sm.json`, which doesn't seem to exist when Preview mode is used. So the props passed to my `SliceZone` component don't include any registries.

`useGetStaticProps` is behaving as expected when not using Preview.

**Comment by hypervillain** [comment]
Thanks for the detailed report! I believe this would require a rewrite, let me talk about it with the team tomorrow and see how we can tackle it

**Comment by AdamDiament** [comment]
@hypervillain  this is also a problem for my Dev team, thanks for looking into it.

**Comment by benjaminsiegal** [comment]
@hypervillain this is also an issue for our team. Thanks for looking into this! Some of our clients are waiting for this to be resolved before moving to Prismic.

**Comment by hypervillain** [comment]
Alright, who wants to try `next-slicezone@0.0.12-alpha.0`? 😊

The only change to your code it requires would be to add a condition on when the resolver is being called in `pages/_document`. Mine is like this at the moment:

```javascript
import Document, { Html, Head, Main, NextScript } from 'next/document'

import { createResolver } from 'next-slicezone/resolver'

export default class extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    if (process.env.NODE_ENV === 'development') {
      await createResolver()
    }
    return { ...initialProps }
  }

  render() {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

````

Hope that helps. Let me know!

**Comment by bobsingor** [comment]
@hypervillain Amazing! This is working. Thanks a lot. 

I have a related question. Is it possible to get the Prismic preview bar inside a next.js slicemachine project? We would like to share previewed documents with different department in the company.

**Comment by lihbr** [comment]
Hey @bobsingor, you should be able to enable the toolbar on Next.js independently of Slice Machine, check the snippet here if that can help: https://github.com/prismicio/nextjs-blog/blob/master/components/PrismicScript.js or the dedicated documentation: https://prismic.io/docs/technologies/previews-nextjs ☺️

**Comment by peroh** [comment]
Working for me too! Thanks very much @hypervillain

**Comment by lihbr** [comment]
Merged and published as of `next-slicezone@0.0.12`, closing :tada:
